src/Operation/BulkWrite.php

```diff
@@ -156,6 +294,10 @@ public function __construct(string $databaseName, string $collectionName, array
 
             switch ($type) {
                 case self::INSERT_ONE:
+                    if ($codec) {
+                        $operations[$i][$type][0] = $codec->encode($args[0]);
+                    }
```

Comment:
`is_document()` is used to validate `$args[0]` _before_ the `switch` statement, which means it never checks the result of `DocumentCodec::encode()`. That seems OK because a Document object is certainly going to be considered a document, but I'd suggest making a note of this in a short comment to copy/paste above these calls to `$codec->encode($args[0]);` (or this conditional).

Reply:
Added an explanatory comment.